From 7ec8251e3570decd97b3050c9ba4fa0d59be93e2 Mon Sep 17 00:00:00 2001
From: Morgan Kesler <keslerm@aegiszero.is>
Date: Thu, 13 Oct 2022 15:00:20 -0400
Subject: [PATCH] Fix broken types from latest pilot-link changes

Last year an update to the pilot link source broke this, it's part of
the patch from here https://github.com/desrod/pilot-link/pull/3

This is simply updating the types to match the changes
---
 contact.c       | 2 +-
 jp-contact.c    | 2 +-
 jp-pi-contact.h | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/contact.c b/contact.c
index 5ca3df68..f4fd6be1 100644
--- a/contact.c
+++ b/contact.c
@@ -442,7 +442,7 @@ int copy_address_to_contact(const struct Address *a, struct Contact *c)
    c->advance = 0;
    c->advanceUnits = 0;    
    memset(&(c->birthday), 0, sizeof(struct tm));
-   for (i=0; i<MAX_CONTACT_BLOBS; i++) {
+   for (i=0; i<MAX_BLOBS; i++) {
       c->blob[i] = NULL;
    }
    c->picture = NULL;
diff --git a/jp-contact.c b/jp-contact.c
index 88d7f7f9..1d02ef2c 100644
--- a/jp-contact.c
+++ b/jp-contact.c
@@ -54,7 +54,7 @@ int jp_pack_Contact(struct Contact *c, pi_buffer_t *buf)
    return pack_Contact(c, buf, contacts_v10);
 }
 
-int jp_Contact_add_blob(struct Contact *c, struct ContactBlob *blob)
+int jp_Contact_add_blob(struct Contact *c, Blob_t *blob)
 {
    return Contact_add_blob(c, blob);
 }
diff --git a/jp-pi-contact.h b/jp-pi-contact.h
index f36e679d..320dc76a 100644
--- a/jp-pi-contact.h
+++ b/jp-pi-contact.h
@@ -47,7 +47,7 @@ extern int jp_pack_ContactAppInfo
     PI_ARGS((struct ContactAppInfo *, pi_buffer_t *buf));
 
 extern int jp_Contact_add_blob
-    PI_ARGS((struct Contact *, struct ContactBlob *));
+    PI_ARGS((struct Contact *, Blob_t *));
 extern int jp_Contact_add_picture
     PI_ARGS((struct Contact *, struct ContactPicture *));
 #ifdef __cplusplus